packages/runtime/container-runtime/src/gc/gcHelpers.ts

```diff
@@ -84,3 +98,66 @@ export function generateSortedGCState(gcState: IGarbageCollectionState): IGarbag
 	}
 	return sortedGCState;
 }
+
+/**
+ * Before GC data was written at the root of the summary tree, individual GC blobs were written at each data store;s
+ * snapshot tree. This function consolidates them into the new IGarbageCollectionState format.
+ */
+export async function getSnapshotDataFromOldSnapshotFormat(
```

Comment:
Seems like a great candidate for a pile of unit tests...

Reply:
Agreed

Reply:
It used to be inlined, so not direclty testable as a unit before.  Maybe there are e2e tests that cover this.

Reply:
There was a test covering this - "generates events for nodes that time out on load - old snapshot format" in garbageCollection.spec.ts.
I am adding a unit test for this function explicitly.